insights/specs/datasources/pcp.py

```diff
@@ -0,0 +1,75 @@
+"""
+Custom datasource related PCP (Performance Co-Pilot)
+"""
+import logging
+import datetime
+import os
+
+from insights.core.dr import SkipComponent
+from insights.core.context import HostContext
+from insights.core.plugins import datasource
+from insights.parsers.ros_config import RosConfig
+from insights.combiners.ps import Ps
+from insights.combiners.services import Services
+
+logger = logging.getLogger(__name__)
+
+
+@datasource(Services, HostContext)
+def pcp_enabled(broker):
+    """
+    Returns:
+        bool: True if pmproxy service is on in services
+
+    Raises:
+        SkipComponent: When pmproxy service is not enabled
+    """
+    if not broker[Services].is_on("pmproxy"):
+        raise SkipComponent("pmproxy not enabled")
+    return True
+
+
+@datasource(Ps, RosConfig, HostContext)
+def pmlog_summary_args(broker):
+    """
+    Determines the pmlogger file and the metrics to collect via `pmlog_summary`
+
+    Returns:
+        str: Full arguments string that will be passed to the `pmlog_summary`,
+             which contains the `pmloger` file and the required `metrics`.
+
+    Raises:
+        SkipComponent: Raises when meeting one of the following scenario:
+                       - No pmlogger process is running
+                       - No pmloger file
+                       - No "mandatory on" metrics in `config.ros`
+    """
+    pm_file = None
+    try:
+        ps = broker[Ps]
+        if not ps.search(COMMAND_NAME__contains='pmlogger'):
+            raise SkipComponent("No 'pmlogger' is running")
+
+        pcp_log_date = (datetime.date.today() - datetime.timedelta(days=1)).strftime("%Y%m%d")
+        pm_file = "/var/log/pcp/pmlogger/ros/{0}.index".format(pcp_log_date)
+
+        if not (os.path.exists(pm_file) and os.path.isfile(pm_file)):
+            raise SkipComponent("No pmlogger file: {0}".format(pm_file))
+
+    except Exception as e:
+        raise SkipComponent("Failed to check pmlogger file existence: {0}".format(str(e)))
+
+    metrics = set()
+    try:
+        ros = broker[RosConfig]
+        for spec in ros.specs:
+            if spec.get('state') == 'mandatory on':
```

Comment:
@kgaikwad, @apuntamb - I just checked the latest `config.ros` and did find any `mandatory on` metrics are specified. Please help confirm if it's okay to apply this condition to this spec. Since with the latest `config.ros` nothing will be collected by the pmlogsummary. Thanks.

Reply:
> I just checked the latest config.ros on my test system and did not find any mandatory on metrics are specified. 

@xiangce - did you mean new metrics here? 
For reference - ROS side PR available to add new metrics under `mandatory on` -  https://github.com/RedHatInsights/ros-backend/pull/123. 
In latest config file change also, ROS do have old metrics under `mandatory on`  - https://github.com/RedHatInsights/ros-backend/blob/main/ansible-playbooks/ros_install_and_set_up.yml#L44
so I want to confirm that.

Reply:
Great, Thanks for the confirmation! @kgaikwad